api/src/auth/drivers/oauth2.ts

```diff
@@ -94,6 +95,9 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 				{ code_verifier: payload.codeVerifier, state: generators.codeChallenge(payload.codeVerifier) }
 			);
 			userInfo = await this.client.userinfo(tokenSet);
+			if (validatePayload(env.AUTH_GITHUB_PUBLIC_REGISTRATION_FILTER, userInfo)) {
+				throw new Error('User does not pass registration filter.');
```

Comment:
Shouldn't this be `InvalidCredentialsException`?

Also, I'm impressed how easy this was to implement! I thought there would be more to it.

Reply:
Yeah. I was going to ask what error should be thrown there. I'm impressed at how fast you checked this.

Reply:
No worries. I'd also move the check to after the try/catch otherwise you're catching and re-throwing. 👍